### DISCUSSION TITLE
query joins tables with millions of records in 2 seconds,but it takes only 500ms with MySQL

### DISCUSSION THREAD

**Discussion by soopsle** [apache/doris#32122] [RFC]
v2.1.0  3be
从mysql上搬过来的业务sql，多表join，其中2张超过500w数据
mysql上查询 500ms内
doris查询需要1.5-2s
尝试过 enable_unique_key_merge_on_write，runtime_filter_type，join hint等，变化不大，无法优化到1s内。

```
select ... from 
a(100w)
INNER JOIN b
INNER JOIN c
INNER JOIN d
INNER JOIN e（500w）
INNER JOIN f
INNER JOIN g（500w）
where ...
order ...
limit ...

```

![image](https://github.com/apache/doris/assets/15059588/a45c038f-394d-4c74-9544-e6aeda145f14)

**Comment by yagagagaga** [comment]
可以尝试建索引、colocate join 等优化技术
